### PR TITLE
sarapprover: remove self node cert

### PR DESCRIPTION
**Pull request by mikedanese**
The functionality to bootstrap node certificates is ready but is blocked by a separable issue discussed in: https://github.com/kubernetes/community/pull/1982. The functionality could be useful for power users who want to write their own approvers if the feature could be promoted to beta. In it's current state this feature doesn't help anybody.

I propose that we remove automated approval of node serving certificates for now and work towards getting the node functionality to beta.

cc @awly @kubernetes/sig-auth-pr-reviews 
```release-note
Remove alpha functionality that allowed the controller manager to approve kubelet server certificates.
```

### CODE REVIEW

**Comment by rtripat** [comment]
@mikedanese Will the kubelet continue to have the ability to request a server certificate as an alpha feature?

**Comment by ericchiang** [comment]
+1 from our side

File: pkg/controller/certificates/approver/sarapprove.go [hunk 1]
```diff
@@ -201,28 +192,3 @@ func isSelfNodeClientCert(csr *capi.CertificateSigningRequest, x509cr *x509.Cert
 	}
 	return true
 }
-
```
Comment (rtripat) [inline pkg/controller/certificates/approver/sarapprove.go]: Unrelated to this change but what do you think about adding debug logging in the recognizer's about the exact condition which led to CSR not being approved?

**Comment by mikedanese** [comment]
> @mikedanese Will the kubelet continue to have the ability to request a server certificate as an alpha feature?

The motivation for this change is that the kubelet piece will cease to be alpha, and become beta.

File: pkg/controller/certificates/approver/sarapprove.go (hunk 1 above)
Comment (mikedanese) [reply pkg/controller/certificates/approver/sarapprove.go]: Events are better for exposing reasons to end users. Would this be for certs that are recognized but the subject access review is denied?

File: pkg/controller/certificates/approver/sarapprove_test.go [hunk 2]
```diff
@@ -187,85 +187,13 @@ func TestHandle(t *testing.T) {
...
-}
-
 func TestRecognizers(t *testing.T) {
 	goodCases := []func(b *csrBuilder){
 		func(b *csrBuilder) {
 		},
 	}
 
 	testRecognizer(t, goodCases, isNodeClientCert, true)
-	testRecognizer(t, goodCases, isSelfNodeClientCert, true)
```
Comment (liggitt) [inline pkg/controller/certificates/approver/sarapprove_test.go]: did you mean to remove the self client cert cases?

**Comment by liggitt** [comment]
question on the removed client cert cases, lgtm otherwise

File: pkg/controller/certificates/approver/sarapprove.go (hunk 1 above)
Comment (mikedanese) [reply pkg/controller/certificates/approver/sarapprove.go]: Debug logging might be ok but not all CSRs are expected to be approved by this controller. Unrecognized CSRs are expected under normal circumstances.

**Comment by rbtr** [comment]
@mikedanese Apologies for the zombie thread - I'm trying to follow all the work around Kubelet server cert signing and not quite sure where to ask this -
since the Kubelet can submit a CSR for its server cert, do you think it would be sufficient to automatically approve that Kubelet server cert iff the following can be verified?
- the certificate is shaped like a Kubelet cert
- the requestor is authorized to create a CSR (via SAR)
- the requestor metadata matches the cert Subject and SANs
- that the Subject and SANs represent a known compute instance by querying the cloud provider

I think from reading all the issues and docs around this that the final check is key in that it uses a trusted third party to verify the node metadata/labels that we can't trust the node to set on itself, and we should then be able to approve the CSR. I'm implementing a CSR approving controller to use until the protected node labels work settles and would appreciate your thoughts on this.

**Pull request merged**